profile/solid-webid-profiles.md

```diff
@@ -0,0 +1,295 @@
+# Solid WebID Profiles Spec
+
+**Note:** This is currently the old document.  We can use it as a basis for creating a new one.
+
+## Table of Contents
+
+* [Overview](#overview)
+* [Profile Representation Formats](#profile-representation-formats)
+* [Required Profile Information](#required-profile-information)
+* [Minimum Recommended Profile
+      Information](#minimum-recommended-profile-information)
+    * [Recommendation for User Names in
+      Profiles](#recommendation-for-user-names-in-profiles)
+* [Public and Private Profiles](#public-and-private-profiles)
+    * [Extended Profile](#extended-profile)
+* [Public Key Certificates](#public-key-certificates)
+* [Account Resource Discovery](#account-resource-discovery)
+    * [Storage Discovery](#storage-discovery)
+    * [Inbox Discovery](#inbox-discovery)
+    * [Type Registry Index Discovery](#type-registry-index-discovery)
+
+## Overview
+
+Solid uses WebID Profile Documents for management of user identity and security
+credentials (such as public keys), and user preferences discovery.
+
+## Profile Representation Formats
+
+From the
+[WebID Profile](https://www.w3.org/2005/Incubator/webid/spec/identity/#publishing-the-webid-profile-document)
+spec:
+
+> A WebID Profile Document is an RDF Web resource that MUST be available as
+> [text/turtle](http://www.w3.org/TR/turtle/), but MAY be available in other RDF
+> serialization formats (such as JSON-LD or
+> [HTML+RDFa](http://www.w3.org/TR/rdfa-core/)) if requested through content
+> negotiation.
+
+## Required Profile Information
+
+There are only 3 statements required for a valid (though not very useful)
+WebID Profile Document:
+
+1. Identifying the document as a `foaf:PersonalProfileDocument` instance
+2. Having a `foaf:primaryTopic` predicate
+3. Having that primary topic be a valid `foaf:Agent` type, such as `foaf:Person`
+
+Here's an example of a minimum valid profile, in Turtle (`text/turtle`) format:
+
+```ttl
+@prefix foaf: <http://xmlns.com/foaf/0.1/>.
+<https://alice.databox.com/profile/card>
+    a foaf:PersonalProfileDocument ;
+    foaf:primaryTopic <#me> .
+
+<#me>
+    a foaf:Person .
+```
```

Comment:
This is incorrect.

The only requirements for a WebID Profile Document are the following:

1. It must be available as `text/turtle`
    >  A WebID Profile Document is a Web resource that MUST be available as text/turtle...
    > -- https://www.w3.org/2005/Incubator/webid/spec/identity/#introduction
2. It must conform to one of the [Cool URIs solution for identifying real world objects](https://www.w3.org/TR/cooluris/#solutions) 
    > For WebIDs with fragment identifiers (e.g. #me), the URI without the fragment denotes the Profile Document. For WebIDs without fragment identifiers an HTTP request on the WebID MUST return a 303 with a Location header URI referring to the Profile Document.
    > -- https://www.w3.org/2005/Incubator/webid/spec/identity/#terminology
3. The WebID referent must control the Profile Document
    > The WebID HTTP URI must be one that dereferences to a document the user controls.
    > -- https://www.w3.org/2005/Incubator/webid/spec/identity/#the-webid-http-uri


If we want to create a recommendation as to the shape of a `foaf:PersonalProfileDocument` linked to a WebID, then we should be clear about that intent.

In other words, a WebID Profile Document may or not be a foaf:PersonalProfileDocument, which is a good thing since a WebID is extremely useful as a unique identifier for decentralised authorization and authentication. Optimising for authentication and authorization is not necessarily compatible with presenting someone's person in different public and private or semi private documents. Solid is a great tool to do such things but not if we confuse FOAF Personal Profile Documents and WebID Profile Documents.

Reply:
PS: The logical and interesting conclusion one can reach from the above is that an empty document is a perfectly valid WebID Profile Document.

Note that the one kind of statement that is required to make a WebID profile document useful in Solid is one stipulating a trusted OIDC issuer:

```turtle
ex:ahmed solid:oidcIssuer ex:issuer1 .
```

Note also that if the Solid-OIDC spec was slightly ammended to allow either/or header/representation instead of a MUST on the RDF representation bearing the trusted issuer, then we could get away with simply a header, rendering WebID Profile Document representation requirements void (which might not be a bad thing). See also: https://solid.github.io/solid-oidc/#webid-issuer-discovery.

Reply:
@TallTed, as far as I understand it is the most authoritative document on the subject of WebIDs, is there anything superseding it?